mapie/time_series_regression.py

```diff
@@ -0,0 +1,312 @@
+        """
+        Compare to the method ``fit`` of ``MapieRegressor``, the ``fit`` method
+        of ``MapieTimeSeriesRegressor`` computes the ``conformity_scores_``
+        with relative values.
+
+        Returns
+        -------
+        MapieTimeSeriesRegressor
+            The model itself.
+        """
+        self = super().fit(X=X, y=y, sample_weight=sample_weight)
+        self.conformity_scores_ = self._relative_conformity_scores(X, y)
+        return self
+
+    def partial_fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+    ) -> MapieTimeSeriesRegressor:
+        """
+        Update the ``conformity_scores_`` and ``k_`` attributes when new data
+        with known labels are available.
+        Note: Don't use ``partial_fit`` with samples of the training set.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples_test, n_features)
+            Input data.
+
+        y : ArrayLike of shape (n_samples_test,)
+            Input labels.
+
+        Returns
+        -------
+        MapieTimeSeriesRegressor
+            The model itself.
+
+        Raises
+        ------
+        ValueError
+            If the lenght of y is greater than the lenght of the training set.
+        """
+        X = cast(NDArray, X)
+        y = cast(NDArray, y)
+        if len(X) > len(self.conformity_scores_):
+            raise ValueError(
+                "You try to update more residuals than there are!"
+            )
+        new_conformity_scores_ = self._relative_conformity_scores(X, y)
+        self.conformity_scores_ = np.roll(
+            self.conformity_scores_, -len(new_conformity_scores_)
+        )
+        self.conformity_scores_[-len(new_conformity_scores_):] = (
+            new_conformity_scores_
+        )
+        return self
+
+    def _beta_optimize(
+        self,
+        alpha: Union[float, NDArray],
+        upper_bounds: NDArray,
+        lower_bounds: NDArray,
+    ) -> NDArray:
+        """
+        ``_beta_optimize`` offers to minimize the width of the PIs, for a given
+        difference of quantiles.
+
+        Parameters
+        ----------
+        alpha: Union[float, NDArray]
+            The quantiles to compute.
+        upper_bounds: NDArray
+            The array of upper values.
+        lower_bounds: NDArray
+            The array of lower values.
+
+        Returns
+        -------
+        NDArray
+            Array of betas minimizing the differences
+            ``(1-alpa+beta)-quantile - beta-quantile``.
+
+        Raises
+        ------
+        ValueError
+            If lower and upper bounds arrays don't have the same shape.
+        """
+        if lower_bounds.shape != upper_bounds.shape:
+            raise ValueError(
+                "Lower and upper bounds arrays should have the same shape."
+            )
+        alpha = cast(NDArray, alpha)
+        betas_0 = np.full(
+            shape=(len(alpha), len(lower_bounds)),
```

Comment:
I would modify the shape of `betas_0` to `(n_samples, n_alpha)` to be consistent with `y_pred_low` and `y_pred_up`.

Reply:
Done